### ISSUE TITLE
Touch ID Should be reset when finger prints has changed

### ISSUE THREAD

**Issue by janehlers** [strongbox-password-safe/Strongbox#251]
I sometimes let friends, family and coworkers open my iPhone by giving them the lock screen code, but if they have that they can change fingerprints and thereby open Strongpass. Therefore open with Touch Id must be disabled when fingerprints prints has changed. My homebank and National Id system has that feature.

**Comment by mmcguill** [comment]
That's a great idea. I didn't think it was possible but your query/examples made me do a quick check and it seems possible! 

https://stackoverflow.com/questions/38397477/touchid-detect-new-fingerprints-added-when-does-evaluatedpolicydomainstate-c

Hopefully an easy one to implement too... Thanks!

**Comment by georgesnow** [comment]
some other notes I found about it:

https://swiftrocks.com/detecting-touchid-fingerprint-changes.html

this developer had some interesting notes that info could be "unstable" according to Apple (no idea what that actually means), but seems like there is fairly reliable way to implement it

**Comment by mmcguill** [comment]
Thanks @georgesnow - I don’t think it’s such a big issue here, as long as it only used for Database Unlock (not App Lock).

If the old state changes, we simply ask the user to re enter their credentials and store again... it’s a good idea in general to require full master credentials entry periodically either way (something I hope to bring as a feature soon)

**Comment by mmcguill** [comment]
This feature is available with 1.45.6 which was released today. I've done my own testing, but would be nice to have a few other datapoints to confirm this is working as expected...

**Comment by janehlers** [comment]
Hi
I just tested on my iPad and it works great. Thank you for adding this so quickly.

Kind regards 
Jan Milter Ehlers

**Comment by mmcguill** [comment]
Excellent, glad to hear it, thanks for the suggestion...

**Comment by dan-el** [comment]
The reset just happened to me this morning even though I did not change/add/remove fingerprints.

Steps I did:
1) opened Strongbox on my iPhone, decrypted with Touch ID
2) went to the Chrome App and navigated to github.com for login
3) I intended to use Auto-Type but I then received the prompt that I needed my master pwd again after a biometric change
4) used app switch (double tap home button) to go back to the still active, but already locked Strongbox app. I was still able to use my Touch ID to unlock it again.
5) Only after quitting Strongbox completely and restarting it I had to enter my master pwd upon opening the app.
6) Then I quit and restarted the Chrome App, tried Auto-Type again and that feature still wanted my master pwd, even though I had just entered it in Strongbox before.

Questions:
1) Why did it ask for my master pwd even though I did not change my fingerprints?
2) Why is Auto-Type and the regular App treated differently in that regard?
M
